scipy/spatial/transform/rotation.py

```diff
@@ -131,54 +132,58 @@ def as_dcm(self):
             return dcm
 
     @classmethod
-    def from_dcm(cls, mat, scalar=True):
-        # For now, mat must be a 3D matrix
-        num_rotations = mat.shape[0]
-        if scalar:
-            quat = np.empty((num_rotations, 4))
-            for rot_num in range(num_rotations):
-                A = mat[rot_num]
-                trA = A.trace()
-                choice = np.argmax(np.array([A[0, 0], A[1, 1], A[2, 2], trA]))
-
-                if choice in [0, 1, 2]:
-                    i = choice
-                    j = (i + 1) % 3
-                    k = (j + 1) % 3
-                    quat[rot_num, i] = 1 - trA + 2 * A[i, i]
-                    quat[rot_num, j] = A[j, i] + A[i, j]
-                    quat[rot_num, k] = A[k, i] + A[i, k]
-                    quat[rot_num, 3] = A[k, j] - A[j, k]
-                else:
-                    quat[rot_num] = np.array([
-                        A[2, 1] - A[1, 2],
-                        A[0, 2] - A[2, 0],
-                        A[1, 0] - A[0, 1],
-                        1 + trA])
+    def from_dcm(cls, dcm):
+        """Initialize rotation from direction cosine matrix.
+
+        This classmethod return a `Rotation` object from the input direction
+        cosine matrices. If the input matrix is not orthogonal, this method
+        creates the best approximation of the rotation.
```

Comment:
Just a note for future: I wouldn't called it "best" and the reference to the paper is the most appropriate here.

Reply:
The paper currently lives in my google drive folder, which I don't think is appropriate to include here as a link. Can you give me a link to your original source of the paper? I'll put it here.

Reply:
The journal information at the top of the first page. 

As how to write references, see for example here https://github.com/scipy/scipy/blob/dfb0cc2/scipy/interpolate/_cubic.py#L352-L770